projects/tests/test_create_delete_environments.py

```diff
@@ -0,0 +1,132 @@
+        User.objects.create_superuser(test_superuser["username"], test_superuser["email"], test_superuser["password"])
+        self.app = Apps.objects.create(name="Some App", slug="someapp")
+        self.env_to_be_deleted = Environment.objects.create(
+            app=self.app, name="env-to-be-deleted", project=self.project
+        )
+        self.client.login(username=test_user["email"], password=test_user["password"])
+
+    def test_environment_creation_regular_user(self):
+        """
+        Test regular user not allowed to create environment
+        """
+        n_envs_before = Environment.objects.count()
+        response = self.client.post(
+            f"/projects/{self.project.slug}/createenvironment/",
+            {
+                "environment_name": "new-env-user",
+                "environment_repository": "n",
+                "environment_image": "n",
+                "environment_app": "n",
+                "app": self.app.pk,
+            },
+        )
+        self.assertEqual(response.status_code, 403)
+        n_envs_after = Environment.objects.count()
+        self.assertEqual(n_envs_before, n_envs_after)
+
+    def test_environment_deletion_regular_user(self):
+        """
+        Test regular user not allowed to delete environment
+        """
+        n_envs_before = Environment.objects.count()
+        response = self.client.post(
+            f"/projects/{self.project.slug}/deleteenvironment/", {"environment_pk": self.env_to_be_deleted.pk}
+        )
+        self.assertEqual(response.status_code, 403)
+        n_envs_after = Environment.objects.count()
+        self.assertEqual(n_envs_before, n_envs_after)
+
+
+class EnvironmentTestCaseSuperUser(TestCase):
+    def setUp(self):
+        self.user = User.objects.create_user(test_user["username"], test_user["email"], test_user["password"])
+        self.project = Project.objects.create_project(name="test-env", owner=self.user, description="")
+        User.objects.create_superuser(test_superuser["username"], test_superuser["email"], test_superuser["password"])
+        self.app = Apps.objects.create(name="Some App", slug="someapp")
+        self.env_to_be_deleted = Environment.objects.create(
+            app=self.app, name="env-to-be-deleted", project=self.project
+        )
+        self.client.login(username=test_superuser["email"], password=test_superuser["password"])
+
+    def test_environment_creation_superuser(self):
+        """
+        Test superuser is allowed to create environment
+        """
+        n_envs_before = Environment.objects.count()
+        response = self.client.post(
+            f"/projects/{self.project.slug}/createenvironment/",
+            {
+                "environment_name": "new-env-superuser",
+                "environment_repository": "n",
+                "environment_image": "n",
+                "environment_app": self.app.pk,
+            },
+        )
+        self.assertEqual(response.status_code, 302)
```

Comment:
If I'm not mistaken, you can do this so that post request follows redirect and then we can check for the status code 200
```suggestion
        response = self.client.post(
            f"/projects/{self.project.slug}/createenvironment/",
            {
                "environment_name": "new-env-superuser",
                "environment_repository": "n",
                "environment_image": "n",
                "environment_app": self.app.pk,
            },
            follow=True
        )
        self.assertEqual(response.status_code, 200)
```